### PR TITLE
Tokenizer rewrite

### PR DESCRIPTION
**Pull request by Myles-Trevino**
This is a rewrite of the Tokenizer page in response to issue #837. The page no longer throws errors on switching pages of the DTM table and does not freeze on large datasets. All functionality is retained except the display of the averages and visualization of the "Documents as Rows, Terms as Columns" data orientation option.

The data orientation was not retained as it appears to be impossible to display many columns efficiently with the current tables library as columns cannot be paginated while rows can. The CSV download however still reflects the data orientation option change. To signify this, "(Applied On Download)" text was added next to the data orientation option.

The display of averages was not retained as calculating the averages noticeably reduces performance when navigating the table and can be done in various other software once the CSV is downloaded.

### CODE REVIEW

File: lexos/views/tokenizer_view.py [hunk 1]
```diff
@@ -1,579 +1,197 @@
...
+# This is a flask blueprint. It helps us to manage groups of views. See here
+# for more detail:
 # http://exploreflask.com/en/latest/blueprints.html
 # http://flask.pocoo.org/docs/0.12/blueprints/
-tokenizer_blueprint = Blueprint('tokenizer', __name__)
+tokenizer_blueprint = Blueprint("tokenizer", __name__)
+
+
+def get_session_dtm_options(default_orientation: bool =
+                            True) -> Dict[str, object]:
```
Comment (chantisnake) [inline lexos/views/tokenizer_view.py]: When are we starting to use dict now? I remember we are using named tuples.

File: lexos/views/tokenizer_view.py [hunk 2]
```diff
@@ -1,579 +1,197 @@
...
-                row = "<tr>"
-                for s in l:
-                    row += "<td>" + str(s) + "</td>"
-                row += "</tr>"
-                rows += row
-        # Catch instances where there is no active document (triggers the error
-        # modal)
+
+            # Get the DTM as a list of tuples
+            dtm = get_dtm_matrix(get_session_dtm_options(), file_manager)
```
Comment (chantisnake) [inline lexos/views/tokenizer_view.py]: why are we not using matrix model here?

**Comment by chantisnake** [comment]
I think this is a great improvement. I am really glad it is done, but I am not sure this is the best we can do.

File: lexos/views/tokenizer_view.py (hunk 1 above)
Comment (chantisnake) [reply lexos/views/tokenizer_view.py]: Also, don't forget to put documentation on every function.

**Comment by Weiqi97** [comment]
@chantisnake Thank you for reviewing and I do agree with you. I communicated with @Myles-Trevino earlier. I have an implementation on the Tokenizer model from last summer which did not include the `ServerSide` functionality. I am currently working on it and Myles's implementation has been very helpful.

File: lexos/views/tokenizer_view.py (hunk 2 above)
Comment (Weiqi97) [reply lexos/views/tokenizer_view.py]: My bad. I have not talked about the MVC structure yet. I will try to give the team a quick introduction on it tomorrow.

**Comment by Weiqi97** [comment]
Most issues discussed here are addressed in #873

**Pull request closed without merging**